### PR TITLE
Add related model label in delete blockers

### PR DESCRIPTION
**Pull request by realVinayak**
Fixes https://github.com/specify/specify7/issues/4005

### CODE REVIEW

**Comment by realVinayak** [comment]
See https://github.com/specify/specify7/issues/4005#issuecomment-1718495574

**Comment by realVinayak** [comment]
image

**Comment by grantfitzsimmons** [comment]
@realVinayak Can these be separate just like Collection Object Attribute?

**Comment by realVinayak** [comment]
Like it is in production? Based on code, @maxxxxxdlp made a deliberate decision to show these dependent resources as part of parent resource (so collection object instead of collection object attachment).

**Comment by grantfitzsimmons** [comment]
No, dependent resources should be shown as children collapsed under the parent, but the table icon should reflect the related table (Cit) or (Att) rather than showing the full name printed out?

File: specifyweb/frontend/js_src/lib/components/QueryBuilder/Results.tsx [hunk 1]
```diff
@@ -393,7 +400,8 @@ export function useFetchQueryResults({
   const resultsRef = React.useRef(results);
   const handleSetResults = React.useCallback(
     (results: RA<QueryResultRow | undefined> | undefined) => {
-      const filteredResults = results !== undefined ? filterArray(results) : undefined
+      const filteredResults =
+        results !== undefined ? filterArray(results) : undefined;
```
Comment (maxxxxxdlp) [inline specifyweb/frontend/js_src/lib/components/QueryBuilder/Results.tsx:404]: no negated conditions - swap branches instead
or even better, just use `f.maybe(results, filterArray);`

**Review by maxxxxxdlp — APPROVED** [review]
looks good
hope this file wasn't confusing to work with

(when I was writing this, I kept being confused between directRelationship.model/relatedModel and parentRelationship.model/relatedModel as well as relationship label VS table name)

**Comment by realVinayak** [comment]
> hope this file wasn't confusing to work with

LOL very! Looking at the code, I knew it was gonna be convoluted logic given so many branches. Took me some solid time on understanding boundaries. I think the logic could be improved though, as the parent resources should be available via othersidename and relationship name, and looking for `isdependent` fields, while also including string parsing. Only reason I was able to make my way through this was that I worked on https://github.com/specify/specify7/pull/3752

File: specifyweb/frontend/js_src/lib/components/QueryBuilder/Results.tsx (hunk 1 above)
Comment (realVinayak) [reply specifyweb/frontend/js_src/lib/components/QueryBuilder/Results.tsx:404]: the one time it was obvious to use f.maybe, and I didn't :)

**Comment by realVinayak** [comment]
image
image

**Comment by realVinayak** [comment]
To test:
Make sure deletion blockers appear correctly when tried to delete a resource with references. Do it for forms, and for agent merging dialog.

**Review by grantfitzsimmons — CHANGES_REQUESTED** [review]
image

https://dbfordocskufish-v79-dev.test.specifysystems.org/specify/view/agent/1028/
https://dbfordocskufish-issue-4005.test.specifysystems.org/specify/view/agent/1028/

**Review by grantfitzsimmons — APPROVED** [review]
Wow! 😍 

image

Screenshot 2023-09-14 at 6 12 35 PM

image

**Review by maxxxxxdlp — APPROVED** [review]

**Review by melton-jason — APPROVED** [review]

**Review by CarolineDenis — APPROVED** [review]

**Review by bronwyncombs — APPROVED** [review]
👍 
Screenshot 2023-09-15 at 9 23 35 AM

**Pull request merged**